### ISSUE TITLE
Pet In Editor Window

### ISSUE THREAD

**Issue by Unthrottled** [gabrielchl/intellij-pets#1]
I would have started a discussion, but that isn't available.

Would be neat if the pet could wander around the entire window, rather than being confined to a tool window.

**Comment by Unthrottled** [comment]
I'd be willing to help with this, just didn't know if it was tried and you ran into issues.

**Comment by gabrielchl** [comment]
hey, yeah it's a great idea
but i don't believe that it's possible to do in intellij (i think?)
i think similar to the vscode-pets plugin (https://github.com/tonybaloney/vscode-pets/issues/250), intellij doesn't provide any methods to render stuff around the window :'(

**Comment by Unthrottled** [comment]
I mean you can, it's just the code you have to write is terrible. Java swing development is no fun. However where there is a will there is a way. I've written a plugin that displays gifs in window (wherever) and another that renders a sticker on top of the editor. So it is possible. Don't know how easy it would be to get the pet to stay in just the code editor, but it can be put anywhere in the window :) 


https://user-images.githubusercontent.com/15972415/219352033-52ae0cde-8b4a-4a56-a139-5b81d3f600c7.mp4

**Comment by gabrielleecredera** [comment]
wait what?! didn't know that was even possible, I'll have a look into it when i have time :)

**Comment by Unthrottled** [comment]
Yeah, it's just a matter of finding the right parent component & placing it there.

You can use the UI Inspector to help you find stuff
https://plugins.jetbrains.com/docs/intellij/internal-ui-inspector.html

**Comment by jk779** [comment]
I'd love that feature :) @Unthrottled maybe you can link your plugin to get an inspiration where to start with placing? :) Sadly I'm really rusty in java, but maybe a boring weekend for @gabrielchl comes up or something ;) ^^

**Comment by viaSeunghyun** [comment]
Wow. Hi, @Unthrottled 

The video you attached in your comment on this issue is of a plugin you developed yourself, not a publicly available one, right?
If it's not publicly available, that's a bit of a shame. It's certainly piqued my interest, haha!

**Comment by Unthrottled** [comment]
@viaSeunghyun 

No worries fam, it is publicly available. The JetBrains plugin repository is wanting me to do some nonsense (fix internal SDK usage) before I can upload it to the marketplace.

You can find all the plugins here

- https://github.com/ani-memes/AMII?tab=readme-ov-file#installation
- https://github.com/doki-theme/doki-theme-jetbrains?tab=readme-ov-file#installation

**Comment by spoutnikz** [comment]
any chance having the pets wandering across the screen without the dedicated window? that would be so awesome, especially on 13-inch screens where an open tab isn't ideal !

lovely work though thank you !
